src/makielayout/layoutables/legend.jl

```diff
@@ -6,8 +6,14 @@ function layoutable(::Type{Legend},
     topscene = get_topscene(fig_or_scene)
 
     default_attrs = default_attributes(Legend, topscene).attributes
+    attrs = Attributes(kwargs)
+    orientation = get(attrs, :orientation, nothing)
+    if !isnothing(orientation)
+        default_attrs[:tellheight] = lift(==(:horizontal), orientation)
+        default_attrs[:tellwidth] = lift(==(:vertical), orientation)
+    end
```

Comment:
I think I have done this a bit differently so far with other attributes, where I've set them to `automatic` by default and then checked the other values. Something like
```julia
real_tellwidth = lift(tellwidth, orientation) do tw, o
    tw === automatic ? o == :vertical : tw       
end
``` 

One would then pass `real_tellwidth` to the LayoutObservables at the bottom

Reply:
Ah I see. I've changed it with your suggestion, so now `attrs` stores a value of `tellheight` and `tellwidth` which could be `automatic`, and `LayoutObservables` receives the "computed version" (must be boolean).